src/Castle.ts

```diff
@@ -0,0 +1,218 @@
+import fetch from 'node-fetch';
+import { reduce } from 'lodash';
+import { IncomingHttpHeaders } from 'http2';
+import AbortController from 'abort-controller';
+
+const defaultApiUrl = 'https://api.castle.io';
+
+type TrackParameters = {
+  event: string;
+  user_id: string;
+  user_traits?: object;
+  created_at?: string;
```

Comment:
@nijikon  - do we use timestamp or  created_at as a proper - we need to sync things with ruby and python (other sdks) with namings

Reply:
`created_at` is an optional property that implementer can send to indicate when the event happened, rather than when it was sent. Different from `sent_at`.